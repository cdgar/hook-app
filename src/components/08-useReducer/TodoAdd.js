import React from "react";
import PropTypes from 'prop-types';
import { useForm } from "../../hooks/useForm";

const TodoAdd = ({ handleAddTodo }) => {
  const [{ description }, handleInputChange, reset] = useForm({
    description: "",
  });

  const handleSubmit = (e) => {
    e.preventDefault();

    if (description?.trim()?.length > 1) {
      const newTodo = {
        id: new Date().getTime(),
        desc: description,
        done: false,
      };
      // const actionAdd = { type: "add", payload: newTodo };
      handleAddTodo(newTodo);
      reset();
    }
  };
  return (
    <>
      <h4>Agregar</h4>
      <hr />
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="description"
          className="form-control"
          placeholder="Learn.."
          autoComplete="off"
          value={description}
          onChange={handleInputChange}
        />

        <button type="submit" className="btn btn-primary mt-1 form-control">
          Agregar
        </button>
      </form>
    </>
  );
};

TodoAdd.propTypes = {
    handleAddTodo: PropTypes.func.isRequired
};

export default TodoAdd;
