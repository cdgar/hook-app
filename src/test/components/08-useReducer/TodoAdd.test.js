import React from 'react';
import { shallow } from "enzyme";
import TodoAdd from "../../../components/08-useReducer/TodoAdd";

describe('TodoAdd.js', () => {
    const handleAddTodo = jest.fn()
    const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);

    test('should show TodoAdd', () => {
        expect(wrapper).toMatchSnapshot();
    })    

    test('should not call handleAddTodo', () => {
      const formSubmit = wrapper.find('form').prop('onSubmit');
      formSubmit({ preventDefault(){}});
      expect(handleAddTodo).not.toHaveBeenCalled();
    })

    test('should call handleAddTodo', () => {
      const value = 'todoXD';
      const event = {target: {value: value, name: 'description'}};
      wrapper.find('input').simulate('change', event)
      const formSubmit = wrapper.find('form').prop('onSubmit');
      formSubmit({ preventDefault(){}});
      expect(handleAddTodo).toHaveBeenCalled();
      expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
      expect(handleAddTodo).toHaveBeenCalledWith({id: expect.any(Number), desc: value, done: false});
      expect( wrapper.find('input').prop('value')).toBe('');
    })
    
});