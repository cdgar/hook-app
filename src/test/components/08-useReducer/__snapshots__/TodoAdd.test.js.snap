// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAdd.js should show TodoAdd 1`] = `
<Fragment>
  <h4>
    Agregar
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Learn.."
      type="text"
      value=""
    />
    <button
      className="btn btn-primary mt-1 form-control"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
